nixos/modules/services/web-servers/nginx/cgit.nix

```diff
@@ -0,0 +1,132 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  globalConfig = config;
+  settingsFormat = {
+    type = with lib.types; let
+      value = oneOf [ int str ] // {
+        description = "INI-like atom (int or string)";
+      };
+      values = coercedTo value lib.singleton (listOf value) // {
+        description = value.description + " or a list of them for duplicate keys";
+      };
+
+    in
+    attrsOf (values);
+    generate = name: values:
+      pkgs.writeText name (lib.generators.toKeyValue { listsAsDuplicateKeys = true; } values);
+  };
+in
+{
+  options.services.nginx.virtualHosts = mkOption {
+    type = types.attrsOf (types.submodule ({ config, ... }:
+      let
+        cfg = config.cgit;
+        pathLocationFix = if (cfg.location == "/") then  "" else cfg.location;
```

Comment:
That is way too non-local.  Move this closer to the point of use. Or just remove it altogether as suggested below.

Reply:
I will see for the test afterwards, on the other hand now that the pull request has been updated. how should I do? I keep pushing on my branch? thanks.